docs/source/advanced/apollo-ast.mdx

```diff
@@ -0,0 +1,187 @@
+---
+title: Apollo AST
+---
+
+In order to generate code, Apollo Android reads your schema and operations written in the GraphQL language. To do this,
+we built a GraphQL parser: a tool that can read, interpret, and expose GraphQL language constructs in a way that can be
+used by the library.
+
+Since this parser is not tied to the rest of the library, and as it may be useful for other purposes, we made it
+available as its own artifact that you can use in your projects.
+
+Features:
+
+- Parses Schema and Operations
+- Returns an Abstract Syntax Tree (AST)
+- AST can easily be manipulated (`transform` API)
+- AST can be translated back to valid GraphQL syntax
+- Validation of schema and operations exposing warnings (e.g. use of deprecated fields) and errors
+
+## Add the dependency
+
+Add the `apollo-ast` dependency to your project:
+
+```kotlin:title=build.gradle[.kts]
+dependencies {
+  // ...
+
+  implementation("com.apollographql.apollo3:apollo-ast:x.y.z")
+}
+```
+
+## Parse a document
+
+Use the `parseAsGQLDocument` method to parse a document from a File, a String, or an
```

Comment:
~Technically speaking, `parseAsGQLDocument` also does validation. Maybe we should change the name there.~ 

Oopps no, my bad, it's only parsing 👍